packages/grid/_modules_/grid/hooks/features/columns/gridColumnsUtils.ts

```diff
@@ -100,6 +100,17 @@ export const getGridColDef = (
   if (!type) {
     return columnTypes[DEFAULT_GRID_COL_TYPE_KEY];
   }
+
+  if (!columnTypes[type]) {
+    console.error(
+      [
+        `MUI: The column type "${type}" you are using is not supported.`,
```

Comment:
I wonder if providing the column `field` would allow a developer to resolve the issue faster. Not sure.

Reply:
Another helper could be to print the available types. Such that they can compare the type they used and the correct ones
```js
// Supported types are: "string", "number", "date", "dateTime", "boolean", "singleSelect", "actions"
`Supported types are: ${Object.keys(columnTypes).map(t => `"${t}"`).join(', ')}`
```

Reply:
Is it possible to add types? In the example from the doc, there is not strictly speaking a modification of the available types. It's only a trick to simply colDef reading

> The demo below defines a new column type: usdPrice that extends the native number column type.
```js
const usdPrice: GridColTypeDef = {
  type: 'number',
  width: 130,
  valueFormatter: ({ value }) => valueFormatter.format(Number(value)),
  cellClassName: 'font-tabular-nums',
};
```

```js
<DataGrid
  columns={[
    { field: 'status', width: 130 },
    { field: 'subTotal', ...usdPrice },
    { field: 'total', ...usdPrice },
  ]}
  rows={rows}
/>
```

Reply:
I was also thinking it is possible but it turns out, no -> https://codesandbox.io/s/columntypesgrid-material-demo-forked-hodui?file=/demo.tsx
You can extend existing column types but you can't create a new one. I'm wondering whether we should add this feature?